Still more fexpr fixes

ccasin opened the pull request:
This makes a series of improvements / additions to the fexpr parser and printer.

I tried to organize this branch for easy review:  The first 7 commits each just make an individual change and add a corresponding test case.  They each have a commit message describing the change.  The 8th commit regenerates the parser and test outputs.  The 9th commit runs `make fmt` and regenerates the generated files again.

The changes are:
- Add parsing for the`Empty_array` static constant.
- Add parsing for the `Eq` int comparison operator.
- Add parsing and printing for `Bytes_or_bigstring_set`
- Add parsing and printing for boolean not.
- Add parsing and printing for unary integer arithmetic.
- Handle string literals with escape characters.
- Fix missing "lor" keyword in lexer.

@lukemaurer Sorry these keep coming.  I think I'm now done for the time being.

lukemaurer commented on middle_end/flambda2/parser/flambda_lex.mll:
I'm a bit sad about copying in code from the lexer. Can we possibly use `Scanf.unescape` instead? (We'd still need to be careful about where the string ends, of course.)

ccasin replied:
That's reasonable.  I've force pushed a new version that does as you suggest.  This is a little less general in that it won't handle nested, interleaved comment and string literals in the same way the ocaml parser does, but that's probably fine.

ccasin commented:
@lukemaurer I made the change you suggested.  I also adjusted the history slightly (merging the two commits at the end that regenerated files into one) and then I added one more commit on the end that just fixes an incorrect attribute in one of the tests, which I noticed when regenerating the tests.

mshinwell commented:
@lukemaurer could you work with @ccasin to get this merged please?

lukemaurer commented:
> @lukemaurer I made the change you suggested. I also adjusted the history slightly (merging the two commits at the end that regenerated files into one) and then I added one more commit on the end that just fixes an incorrect attribute in one of the tests, which I noticed when regenerating the tests.

Oh yes, thanks, that's been bothering me ... LGTM

ccasin commented:
I rebased to fix the merge conflict and will hit the green button when the tests pass, as Luke has approved.
